andela/mervels-bn-backend
# #168131410 Setup docker for backend application

### What does this PR do?
add docker setup to backend
#### Description of Task to be completed?
- add `Dockerfile` to build docker container
- add `docker-compose.yml` file to keep configuration for the Postgres database services. 
- add script `npm run docker:start` to run database migration on the Postgres container and start application after
#### How should this be manually tested?
- Clone the repo
- install docker and make sure it's running 
- Run `docker-compose up` command to build and start the application
- Test the endpoint `localhost:3000/api/docs` to display swagger docs from docker container
#### Any background context you want to provide?
- Make sure port 3000 is not used.
- Incase of failing to start on 1st attempt. End the process and run command `docker-compose up` again.
-For windows users make sure docker-compose is installed.
#### What are the relevant pivotal tracker stories?
[#168131410](https://www.pivotaltracker.com/story/show/168131410)
